### PR TITLE
feat: add Datadog Logs output

### LINKED ISSUE #1052: Datadog output using Datadog Logs
**Motivation**

Current Datadog output uses the Datadog Event feature.  
We have a different usage in mind, using Datalog Logs to ingest Falco events.  
It would allow a smoother integration with the Cloud SIEM Datadog feature to centralize all alerts (including falco events). 

**Feature**

Add Datadog output using Logs feature.  
A design may be discussed because a Datadog output already exist.

- Should we create a complete new output for Datadog Logs without any modification to the current Datadog event output?  
- Should we create a new parameter `mode: (event | logs)` within the current Datadog output, with a default mode to event (so no breaking change)?
- Something different?

WDYT?

If we agree on a design, I can work on the implementation.

**Alternatives**

An alternative would be to use the S3 output to store Falco event in S3, then a Datadog forwarder within AWS (using lambda etc) to push falco events to Datadog logs. It is not ideal because it adds some intermediaries and increases the pricing for no real reason.

**Additional context**

### PR DESCRIPTION
**Pull request by yohboy**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area outputs

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1052

**Special notes for your reviewer**:

### CODE REVIEW

File: config_example.yaml [hunk 1]
```diff
@@ -77,6 +77,12 @@ datadog:
   # host: "" # Datadog host. Override if you are on the Datadog EU site. Defaults to american site with "https://api.datadoghq.com"
   # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
 
+datadogLogs:
+  # apikey: "" # Datadog API Key, if not empty, Datadog output is enabled
```
Comment (Issif) [inline config_example.yaml:81]:
```suggestion
datadoglogs:
  # apikey: "" # Datadog API Key, if not empty, Datadog Logs output is enabled
```

Comment (Issif) [reply config_example.yaml:81]: lower case for consistency

File: docs/outputs/datadog_logs.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+|-------------------------------|--------------------------------| -------------------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| `datadogLogs.apikey`          | `DATADOG_LOGS_APIKEY`          |                            | Datadog API Key, if not empty, Datadog output is **enabled**                                                                      |
+| `datadogLogs.host`            | `DATADOG_LOGS_HOST`            | `https://http-intake.logs.datadoghq.com/` | Datadog host. Override if you are on the Datadog EU site                                                                          |
+| `datadogLogs.minimumpriority` | `DATADOG_LOGS_MINIMUMPRIORITY` | `""` (= `debug`)           | Minimum priority of event for using this output, order is `emergency,alert,critical,error,warning,notice,informational,debug or ""` |
+| `datadogLogs.service`         | `DATADOG_LOGS_SERVICE`         | `""`            | The name of the application or service generating the log events. |
```
Comment (Issif) [inline docs/outputs/datadog_logs.md:22]:
```suggestion
| `datadoglogs.apikey`          | `DATADOG_LOGS_APIKEY`          |                            | Datadog API Key, if not empty, Datadog Logs output is **enabled**                                                                      |
| `datadoglogs.host`            | `DATADOG_LOGS_HOST`            | `https://http-intake.logs.datadoghq.com/` | Datadog host. Override if you are on the Datadog EU site                                                                          |
| `datadoglogs.minimumpriority` | `DATADOG_LOGS_MINIMUMPRIORITY` | `""` (= `debug`)           | Minimum priority of event for using this output, order is `emergency,alert,critical,error,warning,notice,informational,debug or ""` |
| `datadoglogs.service`         | `DATADOG_LOGS_SERVICE`         | `""`            | The name of the application or service generating the log events. |
```

File: docs/outputs/datadog_logs.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+> [!NOTE]
+The Env var values override the settings from yaml file.
+
+## Example of config.yaml
+
+```yaml
+datadogLogs:
+  apikey: "" # Datadog API Key, if not empty, Datadog output is enabled
```
Comment (Issif) [inline docs/outputs/datadog_logs.md:31]:
```suggestion
datadoglogs:
  apikey: "" # Datadog API Key, if not empty, Datadog Logs output is enabled
```

File: outputs/datadog_logs.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		c.OTLPMetrics.Outputs.With(attribute.String("destination", "datadogLogs"),
+			attribute.String("status", Error)).Inc()
+		log.Printf("[ERROR] : Datadog Logs - %v\n", err)
+		return
+	}
+
+	go c.CountMetric(Outputs, 1, []string{"output:datadogLogs", "status:ok"})
+	c.Stats.DatadogLogs.Add(OK, 1)
+	c.PromStats.Outputs.With(map[string]string{"destination": "datadogLogs", "status": OK}).Inc()
+	c.OTLPMetrics.Outputs.With(attribute.String("destination", "datadogLogs"),
```
Comment (Issif) [inline outputs/datadog_logs.go:71]:
```suggestion
		go c.CountMetric(Outputs, 1, []string{"output:datadoglogs", "status:error"})
		c.Stats.DatadogLogs.Add(Error, 1)
		c.PromStats.Outputs.With(map[string]string{"destination": "datadoglogs", "status": Error}).Inc()
		c.OTLPMetrics.Outputs.With(attribute.String("destination", "datadoglogs"),
			attribute.String("status", Error)).Inc()
		log.Printf("[ERROR] : Datadog Logs - %v\n", err)
		return
	}

	go c.CountMetric(Outputs, 1, []string{"output:datadogLogs", "status:ok"})
	c.Stats.DatadogLogs.Add(OK, 1)
	c.PromStats.Outputs.With(map[string]string{"destination": "datadoglogs", "status": OK}).Inc()
	c.OTLPMetrics.Outputs.With(attribute.String("destination", "datadoglogs"),
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Review by Issif — APPROVED** [review]
thanks a lot :pray:

**Comment by yohboy** [comment]
Thanks for your review ! 🚀

**Pull request merged**